kong/db/dao/plugins/go.lua

```diff
@@ -36,6 +36,47 @@ do
   end
 end
 
+do
+  local pluginserver_proc
+
+  function go.manage_pluginserver()
+    if pluginserver_proc then
+      -- already being managed
+      return
+    end
+
+    if ngx.worker.id() ~= 0 then
+      -- only one manager
+      pluginserver_proc = true
+      return
+    end
+
+    ngx_timer_at(0, function ()
+      local ngx_pipe = require "ngx.pipe"
+
+      while true do
```

Comment:
We should check to see if `ngx.worker.exiting()` returns `true`, if so then the process should not be respawned.